sympy/physics/quantum/operator.py

```diff
@@ -307,7 +307,7 @@ def _print_contents_latex(self, printer, *args):
 
     def __mul__(self, other):
 
-        if isinstance(other, Operator):
+        if isinstance(other, (Operator, Dagger)):
             return other
```

Comment:
Is there a superclass of `Operator` and `Dagger` that would make more sense here? Should `Dagger` be a subclass of `Operator`?

I don't know the quantum code so well but special casing types like this suggests that things can be better organised somehow.

Reply:
I agree things can be better organized. Also: not every functionality is tested properly.

Reply:
> Is there a superclass of `Operator` and `Dagger` that would make more sense here? Should `Dagger` be a subclass of `Operator`?


 This is exactly what I was thinking and suggested on the mailing list. But while working on it, it turns out if such a change is made `print` doesn't work properly. ` print(Dagger(A))` gives `A`. I think this happens due to calling of `_print_contents` of `Operator`. And due to unfamiliarity with the printing module I found it difficult to further look into it

Reply:
I think that if `Dagger` is a subclass of `Operator` then if you add a `_print_Dagger` method it will take precedence over `_print_Operator`.

Reply:
Apologies for not being able to reply earlier, I tried overriding `_print_contents()` and made a separate `_print()` function for dagger but still it didn't work.

Reply:
> I tried overriding _print_contents() and made a separate _print() function for dagger but still it didn't work

Can you show us the diff? It would be easier to comment after that. I think it should work if you add a new `_print_Dagger()` method.